### PR TITLE
Forms-PreValueSourceType - Additional example added

### PR DESCRIPTION
**Pull request by hfloyd**

### CODE REVIEW

File: Add-ons/UmbracoForms/Developer/Extending/Adding-a-Prevaluesourcetype.md [hunk 1]
```diff
@@ -5,7 +5,13 @@ versionTo: 10.0.0
 
 # Adding a Prevalue Source Type to Umbraco Forms
 
-The following example shows an illustrative custom prevalue source type that returns a hard-coded list of values. It can be extended for your needs via injection of services via the constructor. Dynamic settings can be applied and validated as shown in the [Validate type settings with ValidateSettings()](adding-a-type#validate-type-settings-with-validatesettings) article.
+*This builds on the "[adding a type to the provider model](Adding-a-Type.md)" chapter*
+
+Add a new class to your project and have it inherit from `Umbraco.Forms.Core.FieldPreValueSourceType`, implement the class.
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Extending/Adding-a-Prevaluesourcetype.md:10]:
```suggestion
Add a new class to your project - inherit it from `Umbraco.Forms.Core.FieldPreValueSourceType` and implement the class. 
```

File: Add-ons/UmbracoForms/Developer/Extending/Adding-a-Prevaluesourcetype.md [hunk 2]
```diff
@@ -5,7 +5,13 @@ versionTo: 10.0.0
 
 # Adding a Prevalue Source Type to Umbraco Forms
 
-The following example shows an illustrative custom prevalue source type that returns a hard-coded list of values. It can be extended for your needs via injection of services via the constructor. Dynamic settings can be applied and validated as shown in the [Validate type settings with ValidateSettings()](adding-a-type#validate-type-settings-with-validatesettings) article.
+*This builds on the "[adding a type to the provider model](Adding-a-Type.md)" chapter*
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Extending/Adding-a-Prevaluesourcetype.md:8]:
```suggestion
*This builds on the "[Adding a type to the provider model](Adding-a-Type.md)" article*
```

File: Add-ons/UmbracoForms/Developer/Extending/Adding-a-Prevaluesourcetype.md [hunk 3]
```diff
@@ -72,3 +78,166 @@ namespace MyFormsExtensions
     }
 }
 ```
+
+:::note
+The PreValue model was updated in Umbraco Forms Versions: 8.13.0, 9.5.0, & 10.1.0 to include a `.Caption` property which can be set separately from the `.Value` property. In previous versions, the `Value` would generally be used as the caption when rendered on the form.
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Extending/Adding-a-Prevaluesourcetype.md:83]:
```suggestion
The `PreValue` model in Umbraco Forms Versions 8.13.0, 9.5.0, 10.1.0, and above includes a `.Caption` property which is set separately from the `.Value` property. In the previous versions, the `Value` is generally used as the caption when rendered on the form.
```

File: Add-ons/UmbracoForms/Developer/Extending/Adding-a-Prevaluesourcetype.md [hunk 4]
```diff
@@ -72,3 +78,166 @@ namespace MyFormsExtensions
...
+:::
+
+
+
+### Another Example Using Dependency Injection to Access Additional Services
+
+This example will take a user-provided Content Node and create a custom Prevalue list from property data on that node. Your own `FieldPreValueSourceType` can get its data from wherever you like - an API call, custom functions, etc.
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Extending/Adding-a-Prevaluesourcetype.md:90]:
```suggestion
This example will take a user-provided Content Node and create a custom Prevalue list from the property data on that node. Your own `FieldPreValueSourceType` can get its data from wherever you like - an API call, custom functions, etc.
```

**Review by eshanrnh — COMMENTED** [review]
Looks, good @hfloyd 💪 I've added some minor suggestions and a suggestion to shorten the sentence length. Let me know if it doesn't make sense 😀

**Review by AndyButland — APPROVED** [review]
Looks good to  me, thanks @hfloyd.  Useful to have a more fleshed out example in the docs.

**Review by eshanrnh — APPROVED** [review]
Looks good. Merging it in 💪

**Pull request merged**